core/trino-main/src/main/java/io/trino/server/ServerConfig.java

```diff
@@ -104,4 +121,10 @@ public ServerConfig setQueryInfoUrlTemplate(String queryInfoUrlTemplate)
         this.queryInfoUrlTemplate = Optional.ofNullable(queryInfoUrlTemplate);
         return this;
     }
+
+    public enum CoordinatorShutdownStrategy
+    {
+        CANCEL_RUNNING, // Cancel running queries after grace period has passed
+        ABORT_AFTER_WAITING // Wait for running queries to complete and abort shutdown after grace period has passed
```

Comment:
That doesn't sound very useful, a busy cluster would never get shut down. Why there's no longer an option to wait, but then cancel all remaining queries?